JetAnalyzers/bin/compareJEC.C

```diff
@@ -254,17 +254,21 @@ void compareJEC(string payld1="Winter14_V8", string payld2="",         string pa
 
   int maxTries = 7;
   string strPath;
-  vector<string> paths = {"CondFormats/JetMETObjects/data/"};
-  paths.push_back(string("/fdata/hepx/store/user/aperloff/JEC/80X_Summer16/")+cid1+"/");
-  paths.push_back(string("/fdata/hepx/store/user/aperloff/JEC/80X_Summer16/")+cid2+"/");
-  paths.push_back(string("/fdata/hepx/store/user/aperloff/JEC/80X_Summer16/")+cid3+"/");
-  paths.push_back(string("/home/aperloff/JEC/CMSSW_8_0_20/src/JetMETCorrections/JECDatabase/textFiles/")+cid1+"/");
-  paths.push_back(string("/home/aperloff/JEC/CMSSW_8_0_20/src/JetMETCorrections/JECDatabase/textFiles/")+cid2+"/");
-  paths.push_back(string("/home/aperloff/JEC/CMSSW_8_0_20/src/JetMETCorrections/JECDatabase/textFiles/")+cid3+"/");
+  vector<string> paths = {"","CondFormats/JetMETObjects/data/"};
```

Comment:
I don't like it hard coded either, but files are not necessarily stored in /fdata/hepx/store/user/<username>/JEC/<ERA>/ . I store different JECs are different places and they all have their own patterns of paths. So it cannot be unified, at least not on my workflow. But I made changes so that it is not just for me.

Reply:
1. Remove the commented out lines right below this. If they aren't needed then don't leave them lying around the code.
2. You can follow this SO post to replace your username with the one found by the system. https://stackoverflow.com/questions/8953424/how-to-get-the-username-in-c-c-in-linux That way it will work for whomever is using the code. Also, you can have a command line option to override the search locations, but only if the user specifies something on the command line.

Reply:
I don't think it a good idea to have any preset convention of JEC paths, I sometimes compare my JECs with Andrew's, fixing username in a path does not seem to be useful for myself.

Reply:
Right now you have your username (and mine) hard coded in the the code. Either remove those lines completely or make it configurable. Also, the disk path (/fdata/hepx/) doesn't need to be hard coded. You can make it a command line option with a default. There are so many better ways of doing this than carrying along these hard coded paths.